sda-commons-starter-kafka/src/main/java/org/sdase/commons/spring/boot/kafka/TraceTokenProducerInterceptor.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2022- SDA SE Open Industry Solutions (https://www.sda.se)
+ *
+ * Use of this source code is governed by an MIT-style
+ * license that can be found in the LICENSE file or at
+ * https://opensource.org/licenses/MIT.
+ */
+package org.sdase.commons.spring.boot.kafka;
+
+import java.util.Map;
+import org.apache.kafka.clients.producer.ProducerInterceptor;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.clients.producer.RecordMetadata;
+import org.slf4j.MDC;
+
+public class TraceTokenProducerInterceptor<K, V> implements ProducerInterceptor<K, V> {
+
+  @Override
+  public ProducerRecord<K, V> onSend(ProducerRecord<K, V> record) {
+
+    String mdcTraceToken = MDC.get("Trace-Token");
+    if (mdcTraceToken != null && !mdcTraceToken.isEmpty()) {
+      record.headers().add("Parent-Trace-Token", mdcTraceToken.getBytes());
```

Comment:
NIT: Consider using `getBytes(Charset)`?

Reply:
changed to `mdcTraceToken.getBytes(StandardCharsets.UTF_8)`